autogen/agentchat/groupchat.py

```diff
@@ -203,6 +196,10 @@ def next_agent(self, agent: Agent, agents: Optional[List[Agent]] = None) -> Agen
         if agents is None:
             agents = self.agents
 
+        # Ensure the provided list of agents is a subset of self.agents
+        if not set(agents).issubset(set(self.agents)):
```

Comment:
@joshkyh  sorry to check with you again. Does this subset check makes sense in the this function?